### ISSUE TITLE
Azure: auto-tag user action doesn't tag the creator's email address if the resource is created via the portal 

### ISSUE THREAD

**Issue by swetha-sundar** [cloud-custodian/cloud-custodian#4407]
Summary:
Azure: auto-tag user action doesn't tag the creator's email address if the resource is created via the portal. This happens for resources of almost all types (resource group, Vnets, VMs, Storage Account, etc..)

Steps to reproduce:
1. Create a VM or any resource via Azure portal
2. Run any policy here: https://cloudcustodian.io/docs/azure/examples/autotagusers.html
3. The tag gets attached to the resource but with the creator's email as "Unknown" . No Error is thrown

If you create the resource via the azure cli or sdk, then it correctly tags the user's email address to the resource

**Comment by abhishekpolavarapu** [comment]
Any update on this, I'm facing the same problem both for resources and resource groups created through portal

**Comment by robdyke** [comment]
Experiencing same behaviour from `auto-tag user` action with CreatorEmail set to Unknown on resources created from cli and portal.

Investigating with local cli and ACI.

**Comment by justinhauer** [comment]
@logachev Just tested creating a vm in the portal, and via a different service principle from terraform - so the creator would be that registered app. Both show up as that default 'unknown' after the policy runs.

**Comment by logachev** [comment]
Sorry, I was not keeping track of this issue.. I will take a look today or tomorrow.
Thanks @justinhauer for adding me!

**Comment by robdyke** [comment]
Here is JSON from Operation name: Create or Update Virtual Machine.

You can see that there are a few sources for the CreatorEmail `caller`, `claims/upn` and `claims/name`

```json
{
    "authorization": {
        "action": "Microsoft.Compute/virtualMachines/write",
        "scope": "/subscriptions//resourcegroups/myResourceGroup/providers/Microsoft.Compute/virtualMachines/u18"
    },
    "caller": "first.last@domain.co.uk",
    "channels": "Operation",
    "claims": {
        "aud": "https://management.core.windows.net/",
        "iss": "https://sts.windows.net/def64197-6c74-4d7d-9916-85a01b2fd9b6/",
        "iat": "1576197901",
        "nbf": "1576197901",
        "exp": "1576201801",
        "http://schemas.microsoft.com/claims/authnclassreference": "1",
        "aio": "",
        "http://schemas.microsoft.com/claims/authnmethodsreferences": "pwd",
        "appid": "",
        "appidacr": "2",
        "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/surname": "Dyke",
        "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/givenname": "Rob",
        "groups": "",
        "ipaddr": "",
        "name": "Rob Dyke",
        "http://schemas.microsoft.com/identity/claims/objectidentifier": "",
        "onprem_sid": "",
        "puid": "",
        "http://schemas.microsoft.com/identity/claims/scope": "user_impersonation",
        "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier": "",
        "http://schemas.microsoft.com/identity/claims/tenantid": "",
        "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name": "first.last@domain.co.uk",
        "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/upn": "first.last@domain.co.uk",
        "uti": "XoDES7t91Umu34ahfMcAAA",
        "ver": "1.0"
    },
    "correlationId": "",
    "description": "",
    "eventDataId": "",
    "eventName": {
        "value": "EndRequest",
        "localizedValue": "End request"
    },
    "category": {
        "value": "Administrative",
        "localizedValue": "Administrative"
    },
    "eventTimestamp": "2019-12-13T01:18:38.8266113Z",
    "id": "/subscriptions/resourcegroups/myResourceGroup/providers/Microsoft.Compute/virtualMachines/u18/events/d1e9f47a-ade1-4158-b513-72afbc5594d0/ticks/637117967188266113",
    "level": "Informational",
    "operationId": "4dc85eed-087a-43e9-a16d-2149f45f028d",
    "operationName": {
        "value": "Microsoft.Compute/virtualMachines/write",
        "localizedValue": "Create or Update Virtual Machine"
    },
    "resourceGroupName": "myResourceGroup",
    "resourceProviderName": {
        "value": "Microsoft.Compute",
        "localizedValue": "Microsoft.Compute"
    },
    "resourceType": {
        "value": "Microsoft.Compute/virtualMachines",
        "localizedValue": "Microsoft.Compute/virtualMachines"
    },
    "resourceId": "/subscriptions//resourcegroups/myResourceGroup/providers/Microsoft.Compute/virtualMachines/u18",
    "status": {
        "value": "Succeeded",
        "localizedValue": "Succeeded"
    },
    "subStatus": {
        "value": "",
        "localizedValue": ""
    },
    "submissionTimestamp": "2019-12-13T01:20:02.0722036Z",
    "subscriptionId": "",
    "relatedEvents": []
}
```

**Comment by robdyke** [comment]
```
[robd@rdc cc]$ custodian version
0.8.45.3
```

**Comment by FrederikL85** [comment]
I'm having the same issue - the creator for all my resources, which I've created in the Azure portal are "unknown".

Any update?

**Comment by kapilt** [comment]
@aluong @logachev ^

**Comment by stefangordon** [comment]
Apologies this one took so long to resolve, I have submitted #5298 to fix the immediate issue and filed #5297 for additional improvements.

The event based tagging was working, but manual tagging based on searching activity log history was not working.  The `caller` field was being returned from Azure as `null` due to an incorrect query from us.

**Comment by justinhauer** [comment]
@stefangordon what would the amount of work be to look into #3702, which is loosely related to this?

**Comment by sanjaynaikwadi** [comment]
👍 I am also running in same issue is this been fixed or #5298 is partial fix ? and available in which version ?

**Comment by tomarv2** [comment]
same issue, it used to work until a while ago, this is my policy:

```
policies:
  - name: custom-auto-owner-vm
    resource: azure.vm
    mode:
      type: azure-event-grid
      events: [{
          resourceProvider: 'Microsoft.Compute/virtualMachines',
          event: 'write'
        }]
    filters:
      - "tag:Owner": null
    actions:
      - type: auto-tag-user
        tag: Owner  
```
